### ISSUE TITLE
Inconsistent initialization behavior

### ISSUE THREAD

**Issue by gabrielwalt** [adobe/adobe-client-data-layer#95]
the data layer script does declare the `adobeDataLayer` array and therefore also doesn't initialize it if it hasn't been declared already prior to the script loading.

### Expected Behaviour

It is expected that scripts behave the same way, regardless whether the data layer script is loaded before or after the `adobeDataLayer` variable has been declared.

### Actual Behaviour

The `adobeDataLayer` array must be declared prior to the Adobe Client Data Layer script being loaded, or the data layer doesn't get initialized and won't be useable on the page.

### Reproduce Scenario

Following HTML doesn't fires the `test` event as expected:

```html
<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>Data Layer</title>
    <script src="https://unpkg.com/@adobe/adobe-client-data-layer@1.1.0/dist/adobe-client-data-layer.min.js"></script>
    <script>
        window.adobeDataLayer = window.adobeDataLayer || [];
        window.adobeDataLayer.push({"event": "test", "eventInfo": "hello world"});
        window.adobeDataLayer.push(function(dl) {
            dl.addEventListener("test", function(e) { console.log(e); });
        });
    </script>
</head>
<body>
    <h1>Data Layer</h1>
</body>
</html>
```

But adding `async defer` attributes to the first script element, or interchanging the two `script` elements will make it work.

It would be expected that the data layer script does declare the `adobeDataLayer` array if it hasn't been declared already.

**Comment by onyx-blackbird** [comment]
Small thing to add:
I was trying to integrate this script into our webpack build. Because it then will be in the same js file in the browser I omitted the `window.adobeDataLayer = window.adobeDataLayer || [];` because I thought it will be set anyway by the script. But when you look at the code, it relies on it to be there. So this line of code doesn't make sense, as window.adobeDataLayer will not be defined, even if the script is loaded first. So you can simply use `window.adobeDataLayer = []`.

**Comment by gabrielwalt** [comment]
Thank you, @onyx-blackbird, for your contribution! Your suggested fix makes perfect sense to me.

Realistically, it will probably take 2 weeks at least before we'll be able to address this, but hopefully not much more.

**Comment by bartoszglow** [comment]
Issue fixed as part of #101
